File: [23.12]Design-Pattern/Java/src/behavioral/observer/Adventurer.java
package behavioral.observer;

public class Adventurer implements Observer{
    private String name;

    public Adventurer(String name) {
        this.name = name;
    }

    @Override
    public void receiveNotice(String msg) {
        System.out.println(name +"님 알람이 도착했습니다. 내용: "+ msg);
    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/behavioral/mediactor/Adventurer.kt
package behavioral.mediactor

class Adventurer(val name: String) {
    private var itemMediactor: ItemMediactor? = null

    fun setMediactor(itemMediactor: ItemMediactor) {
        this.itemMediactor = itemMediactor
        itemMediactor.addAdventurer(this)
    }

    fun sendRequestToMediactor(msg: String) {
        itemMediactor?.forwardRequest(msg)
    }

    fun receiveRequestToMediactor(msg: String) {
        println("전달 받은 내용: $msg")
    }

}